util/logger.go

```diff
@@ -42,3 +65,63 @@ func Logger() *RigLogger {
 
 	return logger
 }
+
+// Spin restarts the spinner for a new task.
+func (log *RigLogger) Spin(message string) {
+	if !log.IsVerbose {
+		log.Progress.Spins.Start(message)
+		log.Spinning = true
+	}
+}
+
+// NoSpin stops the Progress spinner.
+func (log *RigLogger) NoSpin() {
+	log.Progress.Spins.Stop()
+	log.Spinning = false
+}
+
+// Info indicates success behavior of the spinner-associated task.
+func (log *RigLogger) Info(format string, a ...interface{}) {
+	if log.IsVerbose || !log.Spinning {
+		log.Channel.Info.Println(fmt.Sprintf(format, a...))
+	} else {
+		log.Progress.Spins.SetMessage(fmt.Sprintf(format, a...))
+		log.Progress.Spins.Succeed()
+	}
+}
+
+// Warning indicates a warning in the resolution of the spinner-associated task.
+func (log *RigLogger) Warning(format string, a ...interface{}) {
+	if log.IsVerbose || !log.Spinning {
+		log.Channel.Warning.Println(fmt.Sprintf(format, a...))
+	} else {
+		log.Progress.Spins.SetMessage(fmt.Sprintf(format, a...))
+		log.Progress.Spins.Warn()
+	}
+}
+
+// Warn is a convenience wrapper for Warning.
+func (log *RigLogger) Warn(format string, a ...interface{}) {
+	log.Warning(format, a...)
+}
+
+// Error indicates an error in the spinner-associated task.
+func (log *RigLogger) Error(format string, a ...interface{}) {
+	if log.IsVerbose || !log.Spinning {
+		log.Channel.Error.Println(fmt.Sprintf(format, a...))
+	} else {
+		log.Progress.Spins.SetMessage(fmt.Sprintf(format, a...))
+		log.Progress.Spins.Fail()
+	}
+}
+
+// Verbose allows Verbose logging of more advanced activities/information.
+// In practice, if the spinner can be in use verbose is a no-op.
```

Comment:
Having a little trouble wrapping my brain around this comment. Is this saying that if the spinner is running verbose output won't be seen? Is that an indicator that using the --verbose flag indicates that the spinner won't run? That is my intuition but I haven't been able to perceive in the code what makes that the case.

Reply:
If you run rig --verbose, you get "classic" logging. The text from Spin() is not output in verbose mode.